### PR TITLE
blocks: use posix_fadvise to tell OS about sequential access and prefetch

### PR DESCRIPTION
**Pull request by marcusmueller**

### CODE REVIEW

File: gr-blocks/lib/file_source_impl.cc [hunk 1]
```diff
@@ -188,7 +188,34 @@ void file_source_impl::open(const char* filename,
...
     // Rewind to start offset
     if (d_seekable) {
-        if (GR_FSEEK(d_new_fp, start_offset_items * d_itemsize, SEEK_SET) == -1) {
+        auto start_offset = start_offset_items * d_itemsize;
+#ifdef _POSIX_C_SOURCE
+#if _POSIX_C_SOURCE >= 200112L
+        // If supported, tell the OS that we'll be accessing the file sequentially
+        // and that it would be a good idea to start prefetching it
+        auto fd = fileno(d_new_fp);
+        static const std::map<int, const std::string> fadv_errstrings = {
```
Comment (daniestevez) [inline gr-blocks/lib/file_source_impl.cc:197]: Just wondering: is there a reason for using this ad-hoc map instead of `strerror_r()`?

**Comment by mbr0wn** [comment]
Does this have a measurable performance effectd?

File: gr-blocks/lib/file_source_impl.cc (hunk 1 above)
Comment (marcusmueller) [reply gr-blocks/lib/file_source_impl.cc:197]: not really, no.

**Comment by marcusmueller** [comment]
> Does this have a measurable performance effectd?

I mainly did this to avoid some strange effects I saw on Reiner's NTP mounts when repeating small files. I haven't actually gotten around benchmarking it on these machines since then, though.

I have to admit that benchmarking on my SSD array doesn't show a measurable advantage. But then again, as soon as your file is larger than a couple of megabytes, prefetching would happen anyway, and on smaller files: well, sooo many confounding factors.

**Review by mormj — APPROVED** [review]

**Pull request merged**